src/player.cpp

```diff
@@ -12124,6 +12072,63 @@ bool player::armor_absorb( damage_unit& du, item& armor )
     return armor.damage >= 5;
 }
 
+void player::bionic_armor_absorb( body_part bp, damage_unit &du ) const
```

Comment:
Just a detail:
Now that I look at it, it would be better to have this function take body part and damage type argument and return armor, similar to how mutation armor does it.